Packages/com.vrchat.ClientSim/Runtime/CameraStacking/ClientSimStackedVRCameraSystem.cs

```diff
@@ -0,0 +1,149 @@
+            {
+                if (IsCameraStackingEnabled)
+                    DestroyCameraStack();
+            }
+            #endif
+        }
+
+        public void InitializeStackedSystem()
+        {
+            #if !UNITY_ANDROID
+            Cameras = new List<Camera>();
+            if (MainSceneCamera != null)
+            {
+                CreateCameraStack();
+                IsInitialized = true;
+                FormattedLog("Camera Stack has been initialized.");
+            }
+            #else
+            gameObject.SetActive(false);
+            #endif
+        }
+
+        private void CreateCameraStack()
+        {
+            for (int i = 0; i < CameraStack.Length; i++)
+            {
+                AddCamera(i);
+            }
+            IsCameraStackingEnabled = true;
+        }
+
+        private void DestroyCameraStack()
+        {
+            for (int i = 0; i < Cameras.Count; i++)
+            {
+                DestroyCamera(i);
+            }
+            IsCameraStackingEnabled = false;
+        }
+
+        private void AddCamera(int index)
+        {
+            GameObject cameraObj = Instantiate(new GameObject(), MainSceneCamera.transform);
+            Camera cam = cameraObj.AddComponent<Camera>();
+            XRDevice.DisableAutoXRCameraTracking(cam, true);
+
+            cam.CopyFrom(MainSceneCamera); // Start by copying all the settings from the main camera
+            #if VRC_VR_STEAM // We only want this on SteamVR.
+            cameraObj.AddComponent<SteamVRCantedProjectionCullingFix>();
+            #endif
+            Cameras.Add(cam);
+
+            cameraObj.tag = "Untagged";
+            cameraObj.name = $"StackedCamera : {CameraStack[index].CameraName}";
+            cam.clearFlags = CameraClearFlags.Depth;
+            cam.depth = 100 - index;
+            cam.cullingMask = CameraStack[index].RenderLayer;
+            cam.useOcclusionCulling = CameraStack[index].UseOcclusionCulling;
+
+            //Remove this cameras layers from the base camera
+            MainSceneCamera.cullingMask = MainSceneCamera.cullingMask ^ CameraStack[index].RenderLayer;
+            FormattedLog($"Created Stacked Camera : {CameraStack[index].CameraName}");
+            
+            // Set the ClientSim UI canvas to use this camera
+            clientSimMenu.SetCanvasCamera(cam);
+        }
+
+        private void DestroyCamera(int index)
+        {
+            Camera cam = Cameras[index];
+            Cameras.RemoveAt(index);
+
+            //Restore Layers from this camera to the main camera
+            MainSceneCamera.cullingMask = MainSceneCamera.cullingMask | CameraStack[index].RenderLayer;
+            Destroy(cam.gameObject);
+        }
+
+        private void FormattedLog(string s)
+        {
+#if UNITY_EDITOR
+            Debug.Log($"[<color=green>VRC Stacked Camera System</color>] {s}");
+#else
+            Debug.Log($"[VRC Stacked Camera System] {s}");
+#endif
```

Comment:
Same question as above, not sure why this would ever run outside of `UNITY_EDITOR`.

Reply:
this is also an artifact from the client's camera stacking logic, though now that I think about it, you're right, even with future VR support, we still won't need to ever run outside of the editor, so I can clean this up!